src/build_queue.rs

```diff
@@ -176,13 +176,41 @@ impl BuildQueue {
         // available one.
         let to_process = match transaction
             .query_opt(
-                "SELECT id, name, version, priority, registry
-                 FROM queue
-                 WHERE attempt < $1
-                 ORDER BY priority ASC, attempt ASC, id ASC
+                "SELECT
+                    id,
+                    name,
+                    version,
+                    priority,
+                    registry
+                 FROM
+                    queue
+                 WHERE
+                    attempt < $1 AND
+                    (
+                        SELECT
+                            CASE
+                                WHEN MAX(b.build_time) IS NULL then TRUE
```

Comment:
We don't create/update a `build` when these failures happen, so this will always just see null.

Reply:
damn you're right